src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1372,10 +1372,12 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
 
         #! ChemicalFormula calls `ase_atoms = atomic_cell.to_ase_atoms(logger=logger)` and `ase_atoms.get_chemical_formula()`
         # Creating and normalizing ChemicalFormula section
-        # TODO add support for fractional formulas (possibly add `AtomicCell.concentrations` for each species)
-        sec_chemical_formula = self.m_create(ChemicalFormula)
-        sec_chemical_formula.normalize(archive, logger)
-        if sec_chemical_formula.m_cache:
-            self.elemental_composition = sec_chemical_formula.m_cache.get(
-                'elemental_composition', []
-            )
+        if any(cell.name == 'AtomicCell' for cell in self.cell):
```

Comment:
This approach seems fine to me, I just generalized in case later we have some case with both atomic and particle cells (although this seems far away at this point)

Reply:
I agree. Keeping it as general as possible for as long as possible will make possible extensions later easier, even if we don't have concrete ideas about them now.